auth/auth.go

```diff
@@ -183,6 +192,15 @@ func (auth *Authenticator) rebuildChannels(princ Principal) error {
 	}
 
 	base.Infof(base.KeyAccess, "Recomputed channels for %q: %s", base.UD(princ.Name()), base.UD(channels))
+
+	if channelsPerUserThreshold := auth.ChannelsPerUser(); channelsPerUserThreshold != nil {
+		channelCount := len(channels)
```

Comment:
I played around with this locally and it looks like `channels` is not the full set of expanded user channels (accessible via roles).

I think you'll have to do this warning after rebuilding channels and roles inside `getPrincipal`.

Reply:
Yeah, that makes sense. it can be moved to getPrincipal.